write_npy takes array by value for seamingly no reason

First of all thanks a lot for writing this crate, it's been very useful to me.

I noticed that `write_npy` takes the array by value, so I can't do this:

```rust
let arr = array![1, 2, 3, 4];
write_npy("foo.npy", &arr);
some_other_function(arr)
```

To me it looks like this is just a typo, since the implementation of `write_npy` only uses a reference to the array. The reference is taken by the `.` in the `array.write_npy(std::fs::File::create(path)?)` call.

I see two solutions to fix this:
* break the API and change the signature of `write_npy` to take `&T` instead of `T`
* implement `WriteNpyExt` for `&ArrayBase` too, although this still won't work for other implementations of `WriteNpyExt`. Maybe it's possible to implement `WriteNpyExt`  for `&T where T:WriteNpyExt`, I'm not sure.

I can try around a bit and submit a pull request for either of thoses options, which do you prefer?

Well, Serde couldn't implement `Serialize` for `T: AsRef<U>` since the `U` is unconstrained:

```rust
impl<T, U> Trait for T
where
    T: AsRef<U>,
{}
```

yields a compiler error:

```
error[E0207]: the type parameter `U` is not constrained by the impl trait, self type, or predicates
 --> src/main.rs:3:9
  |
3 | impl<T, U> Trait for T
  |         ^ unconstrained type parameter
```

It could implement `Serialize` for `T: Deref`, although that would be a somewhat unusual use of `Deref`. (I don't see `Deref` used as a bound very often; it tends to be used just for the convenience of deref coercions.)

Thanks for PR #37.

--- a/src/npy/mod.rs
+++ b/src/npy/mod.rs
@@ -52,10 +52,10 @@ where
 /// # use ndarray_npy::WriteNpyError;
 ///
 /// let arr = array![[1, 2, 3], [4, 5, 6]];
-/// write_npy("array.npy", arr)?;
+/// write_npy("array.npy", &arr)?;
 /// # Ok::<_, WriteNpyError>(())
 /// ```
-pub fn write_npy<P, T>(path: P, array: T) -> Result<(), WriteNpyError>
+pub fn write_npy<P, T>(path: P, array: &T) -> Result<(), WriteNpyError>
 where
     P: AsRef<std::path::Path>,
     T: WriteNpyExt,

